app/components/ChartTypeSelector/index.js

```diff
@@ -21,54 +17,36 @@ class ChartTypeSelector extends Component {
   }
 
   componentWillMount() {
-    if (this.props.type) {
-      this._selectChartType(this.props.type);
+    if (this.props.typeObj.type) {
+      this._selectChartType(this.typeObj.type);
     }
   }
 
-  _selectChartType(type) {
-    // send data to store, already transformed for selected chart type
-    this.props.dispatch(actionTrigger(
-      RECEIVE_CHART_DATA,
-      this.props.transformedData[type]
-    ));
-
-    // Default x() and y() funcs
-    /**
-     * @todo Apply global defaults for chart type and/or data type here
-     */
-    const opts = { type };
-    if (dataIsMultiSeries(this.props.transformedData[type])) {
-      opts.x = multiXY.x;
-      opts.y = multiXY.y;
-    } else {
-      opts.x = singleXY.x;
-      opts.y = singleXY.y;
-    }
-
-    // trigger chart init action if required
-    if (!this.props.type.length) {
-      this.props.dispatch(actionTrigger(RECEIVE_CHART_OPTIONS_INIT, opts));
+  _selectChartType(selectedName) {
+    const typeObj = getChartTypeObject(selectedName);
+    if (typeObj) {
+      // clear error
+      this.props.dispatch(actionTrigger(SELECT_CHART_TYPE, typeObj));
     } else {
-      this.props.dispatch(actionTrigger(RECEIVE_CHART_OPTIONS, opts));
+      // error
```

Comment:
If this is a note for future feature, please style it :`@todo write error logic`

Otherwise remove the else block entirely.